src/common/utils/is-allowed-resource.injectable.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable, lifecycleEnum } from "@ogre-tools/injectable";
+import type { IComputedValue } from "mobx";
+import allowedResourcesInjectable from "../cluster-store/allowed-resources.injectable";
+import type { KubeResource } from "../rbac";
+
+export type IsAllowedResource = (...resources: KubeResource[]) => boolean;
+
+interface Dependencies {
+  allowedResources: IComputedValue<Set<string>>;
+}
+
+const isAllowedResource = ({ allowedResources }: Dependencies) => (
+  (...resource) => {
```

Comment:
Notice: Function solves quirks of caller, which is the wrong place to do it.

E.g. Why should this function handle these scenarios:

- `isAllowedResource(firstResource, secondResource, [thirdResource, fourthResource], [[fifthResource, sixthResource]]);`
- `isAllowedResource();`

Reply:
We'll fix this

Reply:
Resolved with https://github.com/lensapp/lens/pull/4801/commits/7d76753175751bf3d9d42af24a729f985231e857